app/services/voucher_adjustments_service.rb

```diff
@@ -23,42 +23,44 @@ def self.calculate(order)
       handle_tax_included_in_price(order, amount)
     else
       adjustment.amount = amount
+      adjustment.save
     end
-
-    # Move to closed state
-    adjustment.close
   end
 
   def self.handle_tax_excluded_from_price(order, amount)
-    voucher_rate = amount / order.total
+    voucher_rate = amount / order.pre_discount_total
+
+    adjustment = order.voucher_adjustments.first
 
     # Adding the voucher tax part
     tax_amount = voucher_rate * order.additional_tax_total
 
-    adjustment = order.voucher_adjustments.first
     adjustment_attributes = {
-      amount: tax_amount,
       originator: adjustment.originator,
       order: order,
       label: "Tax #{adjustment.label}",
       mandatory: false,
-      state: 'closed',
+      state: 'open',
       tax_category: nil,
       included_tax: 0
     }
-    order.adjustments.create(adjustment_attributes)
+
+    # Update the amount if tax adjustment already exist, create if not
+    tax_adjustment = order.adjustments.find_or_initialize_by(adjustment_attributes)
+    tax_adjustment.amount = tax_amount
+    tax_adjustment.save
```

Comment:
There's a couple of things happening in this method, and maybe it could be clearer if they were split into two methods.

I think the tax adjustment could be moved to a private method, something like:
```suggestion
    update_tax_adjustment_for(adjustment, tax_amount)
```